src/Datagrid/Pager.php

```diff
@@ -43,22 +47,12 @@ class Pager extends BasePager
      */
     public function computeNbResult()
     {
-        $countQuery = clone $this->getQuery();
-
-        if (\count($this->getParameters()) > 0) {
-            $countQuery->setParameters($this->getParameters());
-        }
-
-        if (\count($this->getCountColumn()) > 1) {
-            $this->countCompositePrimaryKey($countQuery);
-        } else {
-            $this->countSinglePrimaryKey($countQuery);
+        if (!$this->getQuery() instanceof ProxyQuery) {
+            throw new \LogicException('Only '.ProxyQuery::class.' is supported');
         }
```

Comment:
@phansys Here we have the same problem we discussed https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1199#pullrequestreview-529410825

The change proposed by @Warxcell (using a paginator) fix the issue for the ProxyQuery, but using `getQueryBuilder` could be considered as a BC-break since it's doesn't exist in the `ProxyQueryInterface` class.